Update mmctl-cli-tool.rst (#3678)
mattermost/docs: source/administration/mmctl-cli-tool.rst

server (e.g., via SSH). This tool is currently in beta and can be used alongside the Mattermost CLI tool.
In the future, the Mattermost CLI tool will be deprecated.

This feature was developed to a large extent by community contributions and we'd like to extend our gratitude to the contributors who have worked on this project. We are currently accepting pull requests for Help Wanted issues

> in the `mattermost-server <https://github.com/mattermost/mattermost-server/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+mmctl>`__ repo. You can learn more about

Reviewer: Not strictly related: this link uses the `Help Wanted` and `Area/mmctl` GitHub labels:

```suggestion
in the `mattermost-server <https://github.com/mattermost/mattermost-server/issues?q=is%3Aissue+is%3Aopen+label%3A%22Help+Wanted%22+label%3AArea%2Fmmctl>`__ repo. You can learn more about
```

.. code-block:: sh

  make build
  
Running the tests
------------------

mmctl has two types of tests: unit tests and end to end tests. To run the unit tests, you just need to execute:

.. code-block:: sh

  make test

> To run the end to end test suite, you need to have a Mattermost server instance running. Check the Developer Setup guide on how to configure a local test server instance.

Reviewer: yep :)

> To run the end to end test suite, you need to have a Mattermost server instance running. Check the Developer Setup guide on how to configure a local test server instance.

Author: ```suggestion
To run the end to end test suite, you need to have a Mattermost server instance running. Check the `Developer Setup guide <https://developers.mattermost.com/contribute/server/developer-setup/ >`_ for instructions around how to configure a local test server instance.
```

> To run the end to end test suite, you need to have a Mattermost server instance running. Check the `Developer Setup guide <https://developers.mattermost.com/contribute/server/developer-setup/ >`_ for instructions around how to configure a local test server instance.

Reviewer 2: ```suggestion
To run the end to end test suite, you need to have a Mattermost server instance running. Check the `Developer Setup guide <https://developers.mattermost.com/contribute/server/developer-setup/>`_ for instructions around how to configure a local test server instance.
```

  make build

Local mode
----------

> Local mode allows platform administrators with access to the Mattermost server to run mmctl commands against the API without needing to have a user registered. To ensure secure usage of this API, the server exposes a local socket that only a user with access to the server's file system can access. This local socket flags the requests coming from the socket, so they can reach the handlers without requiring a user session. 

Reviewer: Small changes to avoid repetition:

```suggestion
Local mode allows platform administrators with access to the Mattermost server to run mmctl commands against the API without needing to have a user registered. To ensure secure usage of this API, the server exposes a local socket that only a user with access to the server's file system can access. The requests coming from the socket are treated as authorized, so they can reach the handlers without requiring a user session.
```

> The API has the same endpoints and mmctl is able to interact with it without needing any modifications. When local mode is enabled a new helper is added to the session. It returns ``true`` if a session is marked as local using the ``--local`` parameter. This helper is used when checking for session permissions to correctly authorize the local sessions.

Reviewer: ```suggestion
The API that the socket exposes follows the same specification that can be found `in the API documentation <https://api.mattermost.com>`, so mmctl is able to interact with it without needing any modifications. When a request comes in through the socket, it is flagged as local by the server, and this flag is taken into account when checking for session permissions to correctly authorize the sessions.
```
